pkg/sql/pg_catalog.go

```diff
@@ -2775,6 +2774,53 @@ CREATE TABLE pg_catalog.pg_type (
 				})
 			})
 	},
+	indexes: []virtualIndex{
+		{
+			partial: false,
+			populate: func(ctx context.Context, constraint tree.Datum, p *planner, db *sqlbase.ImmutableDatabaseDescriptor,
+				addRow func(...tree.Datum) error) (bool, error) {
+
+				h := makeOidHasher()
+				nspOid := h.NamespaceOid(db, pgCatalogName)
+				coid := tree.MustBeDOid(constraint)
+				ooid := oid.Oid(int(coid.DInt))
+
+				// Check if it is a predefined type
+				typ, ok := types.OidToType[ooid]
+				if ok {
+					if err := addPGTypeRow(h, nspOid, typ, addRow); err != nil {
+						return false, err
+					}
+					return true, nil
+				}
+
+				// Check if it is a user defined type
+				descs, err := p.Tables().GetAllDescriptors(ctx, p.txn)
+				if err != nil {
+					return false, err
+				}
+				lCtx := newInternalLookupCtx(descs, db)
+				usrT, ok := lCtx.typDescs[sqlbase.ID(types.UserDefinedTypeOIDToID(ooid))]
+				if ok {
+					usrTyp, err := usrT.MakeTypesT(
+						tree.NewUnqualifiedTypeName(tree.Name(usrT.GetName())),
+						p.makeTypeLookupFn(ctx),
+					)
+					if err != nil {
+						return false, err
+					}
+
+					if err := addPGTypeRow(h, nspOid, usrTyp, addRow); err != nil {
+						return false, err
+					}
+					return true, nil
+				}
+
+				// no errors and matches
```

Comment:
[nit]: Capitalize and end with a period.

Reply:
Done.